src/hotspot/cpu/s390/s390.ad

```diff
@@ -2547,7 +2547,7 @@ operand uimmI8() %{
 //   operand type int
 // Unsigned Integer Immediate: 9-bit
 operand SSlenDW() %{
-  predicate(Immediate::is_uimm8(n->get_long()-1));
+  predicate(Immediate::is_uimm8((julong)n->get_long()-1));
```

Comment:
```suggestion
  predicate(n->get_long() >= 1 && Immediate::is_uimm8((julong)n->get_long()-1));
```

Reply:
I don't think this is necessary. Unsigned subtraction with wrap-around is not undefined behavior.

Reply:
Right, it's not UB, but sometimes it is a bug, and would be flagged by things like -fsanitize=unsigned-integer-overflow, so my preference would be to avoid it if possible.

Reply:
As it is not really required and for `storage to storage` instructions `length = 0` is invalid case, which current code is already taking care of. So I would just simply keep it that way.